### PR TITLE
JS Static tests added to CI (ESLint + JSCS)

### PR DESCRIPTION
**Pull request by Igloczek**
This PR is part of #9089.

I adjusted Travis config files to run `gulp static` task (ESLint + JSCS).
I changed default `target` of `static` task to `file`, b/c by previously all output was saved to XML files, that's not so useful on daily basics.

Test are not green, b/c of reported JS issues.

PS. Upgrading dependencies (#9062) will increase number of reported issues 😉

### CODE REVIEW

File: dev/tools/grunt/configs/eslint.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (omiroshnichenko) [inline dev/tools/grunt/configs/eslint.json:1]: Please, convert jscs config to json too.

File: dev/tools/grunt/tasks/static.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ module.exports = function (grunt) {
         };
 
     grunt.registerTask('static', function (target) {
-        var currentTarget = target || 'test',
+        var currentTarget = target || 'file',
```
Comment (omiroshnichenko) [inline dev/tools/grunt/tasks/static.js:20]: Currently we use `grunt static` to run all test with generating log file and `grunt static:file --file=<path_to_file>` for single file validation. What reason to switch this?

File: dev/tools/grunt/tasks/static.js [hunk 3]
```diff
@@ -26,7 +26,11 @@ module.exports = function (grunt) {
 
         setConfig('eslint', currentTarget, cvf.getFiles(file));
         setConfig('jscs', currentTarget, cvf.getFiles(file));
-        grunt.option('force', true);
+
+        if (grunt.option('force') !== false) {
```
Comment (omiroshnichenko) [inline dev/tools/grunt/tasks/static.js]: Please, revert this changes, we already solve this issue. We add [grunt-continue](https://www.npmjs.com/package/grunt-continue), that provide ability to run in force mode(for collect eslint and jscs errors) and fail on warnings. Internal ticket MAGETWO-67342 now in delivery process.

File: dev/travis/script.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+case $TEST_SUITE in
+    static)
+        TEST_FILTER='--filter "Magento\\Test\\Php\\LiveCodeTest"' || true
+        phpunit -c dev/tests/$TEST_SUITE $TEST_FILTER
+        grunt static --force=false
```
Comment (omiroshnichenko) [inline dev/travis/script.sh]: Please, remove --force.

File: dev/tools/grunt/tasks/static.js (hunk 2 above)
Comment (Igloczek) [reply dev/tools/grunt/tasks/static.js:20]: UX. Dev should be able to quickly run `grunt static` to get some instant, readable feedback, without trying to find undocumented tasks options.

If you need to generate this XML files for some reason, run `gulp static:test` instead.

File: dev/tools/grunt/configs/eslint.json (hunk 1 above)
Comment (Igloczek) [reply dev/tools/grunt/configs/eslint.json:1]: Done.

File: dev/travis/script.sh (hunk 4 above)
Comment (Igloczek) [reply dev/travis/script.sh]: Done.

File: dev/tools/grunt/tasks/static.js (hunk 3 above)
Comment (Igloczek) [reply dev/tools/grunt/tasks/static.js]: Done.

File: dev/tools/grunt/tasks/static.js (hunk 2 above)
Comment (omiroshnichenko) [reply dev/tools/grunt/tasks/static.js:20]: Ok, make sense. But change, please, in `dev/travis/script.sh` `grunt static` to `grunt static:test`.

**Comment by ishakhsuvarov** [comment]
Hi @Igloczek 
I see some review comments unresolved, are you going to continue with this one?
Thanks!

**Comment by Igloczek** [comment]
@ishakhsuvarov I'm just busy as f... Let's keep it open, I'll finish this ASAP.

**Comment by ishakhsuvarov** [comment]
@Igloczek Sure. No rush, just trying to make sure we are on the same page.

File: dev/tools/grunt/tasks/static.js (hunk 2 above)
Comment (Igloczek) [reply dev/tools/grunt/tasks/static.js:20]: But... the whole point is to not use `test` on CI, b/c output is saved only as JUnit XML file, which isn't available, b/c build artifacts are not saved anywhere (even if they will be publicly available, it's way more complicated way than just read the tests output).

Also, `quiet` option do more harm than good, b/c ignore all warnings. 

Results of `gulp static:test`
```
Running "eslint:test" (eslint) task
Warning: Task "eslint:test" failed. Used --force, continuing.
```

Results of `gulp static` aka. `gulp static:file`
```
Running "eslint:file" (eslint) task
/Users/igloczek/Sites/magento2-dev/lib/web/mage/accordion.js
  7:9  error  Parsing error: Unexpected token 'jquery'

✖ 1 problem (1 error, 0 warnings)

Warning: Task "eslint:file" failed. Used --force, continuing.
```

**Review by omiroshnichenko — APPROVED** [review]

**Comment by magento-team** [comment]
@Igloczek Thank you for the contribution
